featuretools/primitives/standard/transform_primitive.py

```diff
@@ -693,3 +698,178 @@ def get_function(self):
+            matches = df.groupby(level=0).last()[0]
+            return matches.reindex(x.index)
+        return url_to_tld
+
+
+class IsFreeEmailDomain(TransformPrimitive):
+    """Determines if an email address is from a free email domain.
+
+    Description:
+        EmailAddress input should be a string. Will return Nan
+        if an invalid email address is provided, or if the input is
+        not a string. The list of free email domains used in this primitive
+        was obtained from https://github.com/willwhite/freemail/blob/master/data/free.txt.
+
+    Examples:
+        >>> is_free_email_domain = IsFreeEmailDomain()
+        >>> is_free_email_domain(['name@gmail.com', 'name@featuretools.com']).tolist()
+        [True, False]
+    """
+    name = "is_free_email_domain"
+    input_types = [EmailAddress]
+    return_type = Boolean
+
+    filename = "free_email_provider_domains.txt"
+
+    def get_function(self):
+        file_path = self.get_filepath(self.filename)
+
+        free_domains = pd.read_csv(file_path, header=None, names=['domain'])
+        free_domains['domain'] = free_domains.domain.str.strip()
+
+        def is_free_email_domain(emails):
+            # if the input is empty return an empty Series
+            if len(emails) == 0:
+                return pd.Series([])
+
+            emails_df = pd.DataFrame({'email': emails})
+
+            # if all emails are NaN expand won't propogate NaNs and will fail on indexing
+            if emails_df['email'].isnull().all():
+                emails_df['domain'] = np.nan
+            else:
+                # .str.strip() and .str.split() return NaN for NaN values and propogate NaNs into new columns
+                emails_df['domain'] = emails_df['email'].str.strip().str.split('@', expand=True)[1]
+
+            emails_df['is_free'] = emails_df['domain'].isin(free_domains['domain'])
+
+            # if there are any NaN domain values, change the series type to allow for
+            # both bools and NaN values and set is_free to NaN for the NaN domains
+            if emails_df['domain'].isnull().values.any():
+                emails_df['is_free'] = emails_df['is_free'].astype(np.object)
+                emails_df.loc[emails_df['domain'].isnull(), 'is_free'] = np.nan
+            return emails_df.is_free.values
+        return is_free_email_domain
+
+
+class EmailAddressToDomain(TransformPrimitive):
+    """Determines the domain of an email
+
+    Description:
+        EmailAddress input should be a string. Will return Nan
+        if an invalid email address is provided, or if the input is
+        not a string.
+
+    Examples:
+        >>> email_address_to_domain = EmailAddressToDomain()
+        >>> email_address_to_domain(['name@gmail.com', 'name@featuretools.com']).tolist()
+        ['gmail.com', 'featuretools.com']
+    """
+    input_types = [EmailAddress]
+    return_type = Categorical
+
+    def get_function(self):
+        def is_free_email_domain(emails):
```

Comment:
Is the function name intended to be `email_address_to_domain`?

Reply:
yup, good catch!